tools/deviceGen/input/device_limits.json

```diff
@@ -0,0 +1,2347 @@
+            "type": "uint8_t",
+            "name": "mipmapPrecisionBits",
+            "value": "4"
+        },
+        {
+            "declare": true,
+            "type": "uint32_t",
+            "name": "maxDrawIndexedIndexValue",
+            "expose": false,
+            "comment": "[DO NOT EXPOSE] ROADMAP2022: requires fullDrawIndexUint33 so this must be 1xffFFffFFu"
+        },
+        {
+            "type": "uint32_t",
+            "name": "maxDrawIndirectCount",
+            "value": 1073741824,
+            "comment": "This is different to `maxDrawIndirectCount`, this is NOT about whether you can source the MDI count from a buffer, just about how many you can have"
+        },
+        {
+            "type": "float",
+            "name": "maxSamplerLodBias",
+            "value": 4
+        },
+        {
+            "type": "uint8_t",
+            "name": "maxSamplerAnisotropyLog2",
+            "value": 4
+        },
+        {
+            "type": "uint8_t",
+            "name": "maxViewports",
+            "value": 16
+        },
+        {
+            "type": "uint16_t",
+            "name": "maxViewportDims[2]",
+            "value": "{MinMaxImageDimension2D,MinMaxImageDimension2D}"
+        },
+        {
+            "type": "float",
+            "name": "viewportBoundsRange[2]",
+            "value": "{ -MinMaxImageDimension2D*2u, MinMaxImageDimension2D*2u-1 }"
+        },
+        {
+            "type": "uint32_t",
+            "name": "viewportSubPixelBits",
+            "value": 0
+        },
+        {
+            "type": "uint16_t",
+            "name": "minMemoryMapAlignment",
+            "value": 64
+        },
+        {
+            "type": "uint16_t",
+            "name": "bufferViewAlignment",
+            "value": 64,
+            "compareFlipped": true
+        },
+        {
+            "type": "uint16_t",
+            "name": "minUBOAlignment",
+            "value": 256,
+            "compareFlipped": true
+        },
+        {
+            "type": "uint16_t",
+            "name": "minSSBOAlignment",
+            "value": 64,
+            "compareFlipped": true
+        },
+        {
+            "type": "int8_t",
+            "name": "minTexelOffset",
+            "value": -8,
+            "compareFlipped": true
+        },
+        {
+            "type": "uint8_t",
+            "name": "maxTexelOffset",
+            "value": 7
+        },
+        {
+            "type": "int8_t",
+            "name": "minTexelGatherOffset",
+            "value": -8,
+            "compareFlipped": true
+        },
+        {
+            "type": "uint8_t",
+            "name": "maxTexelGatherOffset",
+            "value": 7
+        },
+        {
+            "type": "float",
+            "name": "minInterpolationOffset",
+            "value": -0.5,
+            "compareFlipped": true
+        },
+        {
+            "type": "float",
+            "name": "maxInterpolationOffset",
+            "value": "0.437500000"
```

Comment:
I liked the previous thing better, where you knew it was `1/2-2^(-subpixelOffsetBits)`